src/sage/rings/polynomial/multi_polynomial_element.py

```diff
@@ -177,9 +178,14 @@ def __call__(self, *x, **kwds):
             K = x[0].parent()
         except AttributeError:
             K = self.parent().base_ring()
-        y = K(0)
-        for m, c in self.element().dict().items():
-            y += c * prod(v ** e for v, e in zip(x, m) if e)
+        if isinstance(K, TropicalSemiring):
+            y = K.zero()
+            for m, c in self.element().dict().items():
+                y += c * prod(v ** e for v, e in zip(x, m))
+        else:
+            y = K(0)
+            for m, c in self.element().dict().items():
+                y += c * prod(v ** e for v, e in zip(x, m) if e)       
```

Comment:
There should be no need for this change.

Reply:
Do I need to override the `__call__` method?

Reply:
I don't see why. If you need it, provide an example.

Reply:
If I use the code before change,
```
sage: T = TropicalSemiring(QQ, use_min=False)
sage: R.<a,b> = PolynomialRing(T)
sage: p1 = T(3)*a*b + a + T(-1)*b; p1
3*a*b + 0*a + (-1)*b
sage: p1(T(-5),T(-5))
0
```
Even though the result should be $max(3-5-5, 0-5, -1-5)=-5$.

Reply:
That code is removing $a^b \to b \cdot a = 0$ (tropical number on the right). So again, I don't see why it is needed up to initializing `y = K.zero()` (instead of `y = K(0)`, which is different). Sorry I thought I had said the last part, but I realized I deleted it in my rush to comment before my flight.